src/jit/codegenarm64.cpp

```diff
@@ -2722,7 +2722,7 @@ void CodeGen::genLockedInstructions(GenTreeOp* treeNode)
     // The following instruction includes a acquire half barrier
     // TODO-ARM64-CQ Evaluate whether this is necessary
     // https://github.com/dotnet/coreclr/issues/14346
-    getEmitter()->emitIns_R_R(INS_ldaxr, emitTypeSize(treeNode), loadReg, addrReg);
+    getEmitter()->emitIns_R_R(INS_ldaxr, emitActualTypeSize(data), loadReg, addrReg);
```

Comment:
We generally use `emitTypeSize` for loads.  Since it does not force the size to a general purpose register size.  

Here it doesn't effect functionality, but I would prefer to change it to be consistent.